ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+                return <Error>error("no element found for delete");
+            }
+            error? response = self.googleSheetClient->deleteRowByDataFilter(self.spreadsheetId, self.sheetId, filter);
+            if response is error {
+                return <Error> error(response.message());
+            }
+            
+        } else if (deleteKey is map<anydata>) {
+            string metadataValue = self.generateMetadataValue(self.keyFields, deleteKey);
+            sheets:DeveloperMetadataLookupFilter filter = {locationType: "ROW", metadataKey: self.tableName, metadataValue: metadataValue};
+            sheets:ValueRange[]|error rows = self.googleSheetClient->getRowByDataFilter(self.spreadsheetId, self.sheetId, filter);
+            if rows is error {
+                return <Error>error(rows.message());
+            }
+            if (rows.length() == 0) {
+                return <Error>error("no element found for update");
+            }
+            error? response = self.googleSheetClient->deleteRowByDataFilter(self.spreadsheetId, self.sheetId, filter);
+            if response is error {
+                return <Error> error(response.message());
+            }
+
+        }
+    }
+
+    public isolated function getKeyFields() returns string[] {
+        return self.keyFields;
+    }
+
+    public isolated function getManyRelations(record {} 'object, string[] fields, string[] include, typedesc<record {}>[] typeDescriptions) returns Error? {
+        foreach int i in 0 ..< include.length() {
+            string entity = include[i];
+            string[] relationFields = from string 'field in fields
+                where 'field.startsWith(entity + "[].")
+                select 'field.substring(entity.length() + 3, 'field.length());
+
+            if relationFields.length() is 0 {
+                continue;
+            }
+
+            function (record {}, string[]) returns record {}[]|error associationsMethod = self.associationsMethods.get(entity);
+            record {}[]|error relations = associationsMethod('object, relationFields);
+            if (relations is error) {
+                return <Error>error("unsupported data format");
+            }
+            'object[entity] = relations;
+        }
+    }
+
+    public isolated function addKeyFields(string[] fields) returns string[] {
+        string[] updatedFields = fields.clone();
+
+        foreach string key in self.keyFields {
+            if updatedFields.indexOf(key) is () {
+                updatedFields.push(key);
+            }
+        }
+        return updatedFields;
+    }
+
+    private isolated function dataConverter(json value, string dataType) returns time:Date|time:TimeOfDay|time:Civil|time:Utc|int|string|decimal|error {
+        if (dataType == "int") {
+            return int:fromString(value.toString());
+        } else if dataType == "time:Date" || dataType == "time:TimeOfDay" ||dataType == "time:Civil" || dataType == "time:Utc" {
+            
```

Comment:
```suggestion
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)